gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-kafka/src/main/java/io/gravitee/plugin/endpoint/kafka/KafkaEndpointConnector.java

```diff
@@ -98,10 +100,10 @@ protected Completable publish(final ExecutionContext ctx) {
                     // Set kafka producer properties
                     config.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
                     config.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class);
-                    config.put(ProducerConfig.CLIENT_ID_CONFIG, UUID.randomUUID().toString());
+                    config.put(ProducerConfig.CLIENT_ID_CONFIG, KAFKA_CLIENT_IDENTIFIER);
                     addCustomProducerConfig(config);
                     SenderOptions<String, byte[]> senderOptions = SenderOptions.create(config);
-                    KafkaSender<String, byte[]> kafkaSender = KafkaSender.create(CustomProducerFactory.INSTANCE, senderOptions);
+                    KafkaSender<String, byte[]> kafkaSender = KafkaSenderFactory.INSTANCE.createSender(senderOptions);
```

Comment:
The factory should not be a singleton instance but a regular instance stick to the current connector instance because the lifecycle of the connector have direct consequences on the created kafka senders (doStop clears the map). If the factory stays static it means a single map of senders is shared across all the connector instances, and so, restarting an api will clear the map of sender for all the apis (also the kafka client id is the same for all apis, don't think it will work).

Reply:
That is already fixed.